libbeat/cfgfile/list.go

```diff
@@ -85,7 +85,7 @@ func (r *RunnerList) Reload(configs []*reload.ConfigWithMeta) error {
 	for hash, runner := range stopList {
 		r.logger.Debugf("Stopping runner: %s", runner)
 		delete(r.runners, hash)
-		go runner.Stop()
+		runner.Stop() // we need to wait this to finish before starting new inputs
```

Comment:
Do we? It did not fix the problem we had, so I do not want to change this to be blocking. Why "fix" something that is not broken?

Reply:
Indeed.

This PR still a draft of a draft, it still needs cleaning up. If we decide to keep working on that I'll keep it async.